### ISSUE TITLE
Fehler beim Compilieren

### ISSUE THREAD

**Issue by Timmy94W** [AtomSponge/skyblock-mp#1]
Ich erhalten diesen Fehler, wenn ich diese Source compilieren möchte

[code]
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariConfig.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariDataSource.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariConfig.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariDataSource.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariConfig.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariDataSource.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariConfig.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/HikariDataSource.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
warning: C:\Users\timwi.gradle\caches\modules-2\files-2.1\com.zaxxer\HikariCP\2.3.13\e152a22511dfe23bf3fd2b660c96467524553965\HikariCP-2.3.13.jar(com/zaxxer/hikari/AbstractHikariConfig.class): major version 52 is newer than 51, the highest major version supported by this compiler.
  It is recommended that the compiler be upgraded.
E:\GitHub\skyblock-mp\build\sources\java\com\github\atomsponge\skyblockmp\database\DatabaseUpdater.java:71: error: cannot find symbol
        final String[] statements = String.join(" ", lines).trim().replaceAll(" +", " ").replaceAll("\n", " ").split(";");
                                          ^
  symbol:   method join(String,List<String>)
  location: class String
1 error
9 warnings
:compileJava FAILED

FAILURE: Build failed with an exception.
- What went wrong:
  Execution failed for task ':compileJava'.
  
  > Compilation failed; see the compiler error output for details.
- Try:
  Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

[/code]

**Comment by AtomSponge** [comment]
Java 8 wird benötigt - entschuldige die späte Antwort, aus irgendeinem Grund habe ich keine Notification für die Issue bekommen :/

**Comment by Timmy94W** [comment]
Ok, Danke, kein Thema, das mit der Wartezeit. kannst du mir verraten, wo ich die schematic herbekomme?

**Comment by AtomSponge** [comment]
WorldEdit wird zum Importieren der Schematic benötigt, hiermit kannst du diese auch erstellen: http://minecraft.curseforge.com/projects/worldedit

Zu der [Frage auf Twitter](https://twitter.com/Timmy94W/status/737335118038192128):
1. Kompilieren (via Gradle-Task reobf)
2. In den Mods-Ordner des Forge-Servers kopieren, Server einmal starten und direkt wieder stoppen, damit die Datenbank, die Konfigurationen etc. erstellt werden. 
   Die Einstellungen (config/skyblockmp.cfg) sind kommentiert und sollten selbsterklärend sein, das _spawn-offset_ (kannst du dir als Vektor von bspw. x=1000, y=80, z=1000 bei einem _grid spacing_ von 1000 Blöcken und der _island-height_ von 80 Blöcken vorstellen, der zu der Position auf der Insel zeigt, auf der die Spieler später spawnen werden) muss je nach Schematic angepasst werden.
   Für kleinere Server kannst du die Konfigurationen bezüglich der Datenbank so belassen, bei größeren Servern würde ich empfehlen, MySQL/MariaDB/Percona/... anstatt H2 zu nutzen.
3. Die zuvor (entweder vor der Installation von SkyblockMp oder auf einem anderen Server erstellte) Schematic nach skyblock/island.schematic kopieren. Nutzen mehrere Server die selbe Schematic, kannst du den Pfad auch in der Konfiguration anpassen.
4. In den server.properties das _level-type_ auf _SKYBLOCKMP_ ändern.
5. Server wieder starten. Optional kannst du auf der Zentralinsel (x=0, z=0; diese ist "reserviert" und wird nicht an Spieler vergeben) manuell eine Schematic importieren.

Bitte beachte, dass ich die Mod ausschließlich für 1.7.10 entwickelt und getestet habe und momentan kein zeitnahes Update auf 1.8/1.9 geplant ist - wenn du es dringend brauchst, kannst du gern selbst updaten und einen Pull Request erstellen :wink: 

Mit der _/island invite_-Funktion würde ich auf öffentlichen Servern vorsichtig sein, da der eingeladene Spieler (noch) nicht gefragt wird, ob er überhaupt eingeladen werden möchte... Da kann ich aber zeitnah (evtl. am Wochenende) eine Abfrage einbauen. Wenn _general.create-island-on-join_ aktiv ist, spielt dies aber keine Rolle, da nur Spieler ohne Insel eingeladen werden können.

Ich sollte mal eine ReadMe mit groben Installationsanweisungen schreiben... Demnächst bitte englische Issues erstellen, damit andere, nicht deutschsprachige Nutzer auch davon profitieren können, wenn ich die Mod mal irgendwann in einem ausgereifteren Stadium auf CurseForge hochlade.

**Comment by Timmy94W** [comment]
Wenn ich Java könnte, dann würde ich dir gern helfen, aber leider kann ich es nicht und ich habe noch kein gutes Tutorial dafür gefunden. Wenn du eines kennst, dann lasse es mich wissen :)

**Comment by AtomSponge** [comment]
@Bandelenth hat glaube ich gute Erfahrungen mit ["Java - Der umfassende Programmierkurs" von O'Reilly](https://www.oreilly.de/buecher/120065/9783955615406-java---der-umfassende-programmierkurs.html) (gibt es auch bei Amazon), ansonsten trial and error :smile:

**Comment by Timmy94W** [comment]
Kennst du die Scriptsprache PAWN?  Wenn es da Ähnlichkeiten gibt, dann kann man das ja schnell lernen

**Comment by AtomSponge** [comment]
Nein, kenne ich nicht, grundsätzlich kann man Skriptsprachen aber eigentlich nicht mit Programmiersprachen "gleichsetzen".

**Comment by Timmy94W** [comment]
Das ist wahr, aber Scriptsprachen bedienen sich der Elemente der Programmiersprachen.
